jmx/jmx.go

```diff
@@ -361,11 +361,7 @@ func receiveResult(lineC chan []byte, queryErrC chan error, cancelFn context.Can
 			for k, v := range r {
 				result[k] = v
 			}
-			return
 
-		case warn = <-cmdWarnC:
```

Comment:
Is there a reason for defering this? I'm afraid that as a side effect of consuming the channel later, it could lead to cmdWarnC being filled up and blocking potential errors being sent to error channel

Reply:
good point, we can increase queue, of restore this case